fix: added envVars to containers

Retna-Gjensidige opened the pull request:
Signed-off-by: Retna Ramachandran <retna@gjensidige.no>
## Related issue
#2036 
<!--
Please link the GitHub issue this pull request resolves in the format of `#1234`. If you discussed this change
with a maintainer, please mention her/him using the `@` syntax (e.g. `@JimBugwadia`).

If this change neither resolves an existing issue nor has sign-off from one of the maintainers, there is a
chance substantial changes will be requested or that the changes will be rejected.

You can discuss changes with maintainers in the [Kyverno Slack Channel](https://kubernetes.slack.com/).
-->

## What type of PR is this
/kind bug

<!--

> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading white spaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
-->

## Proposed Changes
Added the envVar configuration block to containers.
<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. 

***NOTE***: If this PR results in new or altered behavior which is user facing, you **MUST** read and follow the steps outlined in the [PR documentation guide](pr_documentation.md) and add Proof Manifests as defined below.
-->

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [] I have added tests that prove my fix is effective or that my feature works.
- [] My PR contains new or altered behavior to Kyverno and
  - [] I have added or changed [the documentation](https://github.com/kyverno/website) myself in an existing PR and the link is:
  <!-- Uncomment to link to the PR -->
  <!-- https://github.com/kyverno/website/pull/123 -->
  - [] I have raised an issue in [kyverno/website](https://github.com/kyverno/website) to track the doc update and the link is:
  <!-- Uncomment to link to the issue -->
  <!-- https://github.com/kyverno/website/issues/1 -->
  - [] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.

## Further Comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution
you did and what alternatives you considered, etc...
-->

realshuting reviewed (commented):
@vyankd - should we separate 2 envs?

vyankyGH commented:
@Retna-Gjensidige  Can we separate 2 envs one for Container and initContainer?

Retna-Gjensidige commented:
> @Retna-Gjensidige Can we separate 2 envs one for Container and initContainer?

Sure, if you want to separate it, i can do the changes. 

Proposal:
envVarsInit - for initContainer
envVars - for Container

Is that ok?

vyankyGH commented:
Yes, That will work.

Retna-Gjensidige commented:
> Yes, That will work.

Do want me to create a feature request for this implementation or just continue on this branch/pr?

georgekaz commented on charts/kyverno/templates/deployment.yaml:
```diff
@@ -104,6 +104,10 @@ spec:
           - name: KYVERNO_SVC
             value: {{ template "kyverno.serviceName" . }}
+          {{- with .Values.envVars }}
+          env:
```
env key for this container is already declared on line 98.

Retna-Gjensidige replied:
i'll clean these up with the new split in envVars

vyankyGH commented:
> > Yes, That will work.
> 
> Do want me to create a feature request for this implementation or just continue on this branch/pr?

we can continue with this PR

Retna-Gjensidige commented:
> > > Yes, That will work.
> > 
> > 
> > Do want me to create a feature request for this implementation or just continue on this branch/pr?
> 
> we can continue with this PR
@georgekaz - I have cleaned up the duplicate env keys
@vyankyGH - I have split the env variables

vyankyGH commented:
@Retna-Gjensidige Thanks for the changes. Are this changes will support backward compatibility?

Retna-Gjensidige commented:
> @Retna-Gjensidige Thanks for the changes. Are this changes will support backward compatibility?
These changes are Helm specific and will follow the chart version you would set on each release. 
Only compatibility issue would be an user looking at the new docs but using an older chart. Nothing will break but the intended functionality(envVarsInit) will not be there.
